specifyweb/frontend/js_src/lib/components/Statistics/types.ts

```diff
@@ -13,17 +13,23 @@ export type CustomStat = State<
   }
 >;
 
+export type PartialQueryFieldWithPath = Partial<
+  SerializedResource<SpQueryField>
+> & {
+  readonly path: string;
+};
+
 export type DefaultStat = State<
   'DefaultStat',
   {
-    readonly itemType: 'BackEndStat' | 'QueryStat';
+    readonly itemType: 'BackEndStat' | 'DynamicStat' | 'QueryStat';
     readonly pageName: string;
     readonly categoryName: string;
     readonly itemName: string;
     readonly label: string;
     readonly itemValue: number | string | undefined;
     readonly isVisible?: boolean;
-    readonly pathToValue?: string;
+    readonly pathToValue?: number | string | null;
```

Comment:
please don't use null!

by using nulls anywhere in the code you curse yourself into having to do two checks always or risk having a bug slip in:
```js
resolvedSpec?.pathToValue !== undefined &&
    resolvedSpec?.pathToValue !== null 
```

Reply:
Fixed